src/chia_log/handlers/block_handler.py

```diff
@@ -0,0 +1,51 @@
+# std
+import logging
+from typing import List, Optional
+
+# project
+from . import LogHandler
+from ..parsers.block_parser import BlockParser
+from .condition_checkers import BlockConditionChecker
+from .condition_checkers.found_blocks import FoundBlocks
+from .daily_stats.stats_manager import StatsManager
+from src.notifier import Event, EventService, EventType, EventPriority
+
+
+class BlockHandler(LogHandler):
+    """This handler parses all logs indicating found block
+    activity by the full node. It holds a list of condition checkers
+    that are evaluated for each event.
+    """
+
+    def __init__(self):
+        self._parser = BlockParser()
+        self._cond_checkers: List[BlockConditionChecker] = [FoundBlocks()]
+
+    def handle(self, logs: str, stats_manager: Optional[StatsManager] = None) -> List[Event]:
+        """Process incoming logs, check all conditions
+        and return a list of notable events.
+        """
+
+        events = []
+        activity_messages = self._parser.parse(logs)
+        if stats_manager:
+            stats_manager.consume_block_messages(activity_messages)
+        
+        # Create a keep-alive event if any logs indicating
+        # activity have been successfully parsed
+        if len(activity_messages) > 0:
+            logging.debug(f"Parsed {len(activity_messages)} activity messages")
+            events.append(
+                Event(
+                    type=EventType.KEEPALIVE, priority=EventPriority.NORMAL, service=EventService.FULL_NODE, message=""
+                )
+            )
```

Comment:
The keep-alive monitor is currently not taking into consideration events from any other service than HARVESTER. Although that's something we might want to explore in the future, I think we should keep it separate from this PR and remove that event (same as FinishedSignagePointHandler).